test/images/agnhost/webhook/pods.go

```diff
@@ -93,11 +104,9 @@ func mutatePods(ar v1.AdmissionReview) *v1.AdmissionResponse {
 	}
 	reviewResponse := v1.AdmissionResponse{}
 	reviewResponse.Allowed = true
-	if pod.Name == "webhook-to-be-mutated" {
```

Comment:
unconditionally adding an init container is not idempotent (will fail if the webhook is rerun), so the existing code looks incorrect

Reply:
cc @jpbetz @roycaihw

Reply:
> unconditionally adding an init container is not idempotent (will fail if the webhook is rerun), so the existing code looks incorrect

I'm not sure I understand... I'm removing pod name matching so it should be unrelated to webhook being rerun. Could you give me an example of a scenario?
Plus, patch that it's used overrides existing initContainers, so result should always be the same...

Reply:
The existing code always appends a container to the pod being admitted. That is not idempotent (if the webhook is called twice while admitting the pod, it will append two identically named containers, which will fail validation). It should only append the container if the named injected container does not already exist.

Reply:
Ok, I understand that I need to add this check for sidecar injection, so thanks for  point it out :)

But the adding initContainer should still work fine. The patch used for initContainers is `"{"op":"add","path":"/spec/initContainers","value":[{"image":"webhook-added-image","name":"webhook-added-init-container","resources":{}}]}"`, which should override the list.
Imho `{"op":"add","path":"/spec/initContainers/-","value":{"image":"webhook-added-image","name":"webhook-added-init-container","resources":{}}}` should add an init container to the existing list.

Reply:
> But the adding initContainer should still work fine. The patch used for initContainers is "{"op":"add","path":"/spec/initContainers","value":[{"image":"webhook-added-image","name":"webhook-added-init-container","resources":{}}]}", which should override the list.

That is not well-behaved as a webhook... it means if a subsequent admission webhook appended an initContainer, then this webhook was called again, it would undo the other webhook's change. See https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#idempotence

Reply:
`mutatePod' should have unchanged logic now. I also added check for the sidecar so the webhook should be idempotent.
PTAL